draft-ietf-tls-esni.md

```diff
@@ -456,6 +456,9 @@ The client then constructs the ClientHelloInner message just as it does a
 standard ClientHello, with the exception of the following rules:
 
 1. It MUST NOT offer to negotiate TLS 1.2 or below.
+1. It MUST ensure that all HelloRetryRequest-sensitive parameters applicable
```

Comment:
That seems like an improvement, yeah. I'll take it.

Reply:
Is it parameters that would change in response to HRR, or parameters that would change which HRRs you accept? The latter captures the issue a bit more directly, since the issue is a goofy mechanical one of needing to process an HRR for two ClientHellos concurrently, and the error cases being a mess. (If the HRR matches inner but not outer, how do you construct a new outer ClientHello?)

Also a strict reading of "parameters that might change in response to receiving HRR" would include PSKs, which isn't what we want. RFC8446 has this odd corner case:
https://tools.ietf.org/html/rfc8446#section-4.1.2

>  -  Updating the "pre_shared_key" extension if present by recomputing
>      the "obfuscated_ticket_age" and binder values and (optionally)
>      removing any PSKs which are incompatible with the server's
>      indicated cipher suite.

(This was to save clients from recomputing binders for a hash they aren't using. Something something crypto agility is a headache.)

A strict reading would also exclude cipher suite preferences, since the client doesn't change them on HRR, even though the server has already committed to one.

Reply:
I applied the suggestion above!